### PR TITLE
feat(rds): Add AWS RDS cluster transport encryption check

### PR DESCRIPTION
**Pull request by madereddy**
### Context

Add additional RDS cluster transport level encryption logic for supported RDS versions:

 >   For PostgreSQL and Aurora PostgreSQL clusters, if the rds.force_ssl parameter value is set to 0, the Transport Encryption feature is not enabled. For MySQL, Aurora MySQL and MariaDB clusters, if the require_secure_transport parameter value is set to OFF, the Transport Encryption feature is not enabled.

### Description

Added checks for MySQL, MariaDB, PostgreSQL, Aurora PostgreSQL, and Aurora MySQL DB clusters.

Had to modify rds_instance_deletion_protection check and test as well to deal the modification to the db_clusters which allows the parameters to be read.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by madereddy** [comment]
Competing with https://github.com/prowler-cloud/prowler/pull/4002 and https://github.com/prowler-cloud/prowler/pull/4003

Recommended to merge this after https://github.com/prowler-cloud/prowler/pull/4002 as that is a more important check. Cert: rds-ca-2019 will be expiring August 22nd 2024.

File: prowler/providers/aws/services/rds/rds_service.py
Comment (jfagoagas) [inline prowler/providers/aws/services/rds/rds_service.py:17]: This needs to be kept as a dict using the cluster ARN as key.

File: prowler/providers/aws/services/rds/rds_service.py
Comment (jfagoagas) [inline prowler/providers/aws/services/rds/rds_service.py:199]: Why this?

File: prowler/providers/aws/services/rds/rds_service.py
Comment (jfagoagas) [inline prowler/providers/aws/services/rds/rds_service.py:173]: This needs to be kept as it was.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/aws/services/rds/rds_service.py
Comment (madereddy) [reply prowler/providers/aws/services/rds/rds_service.py:17]: Changed this to a list so that it would match self.db_instances and could use similar parameter group code. I can switch this back to a dict if required or if it would cause breaking changes.

File: prowler/providers/aws/services/rds/rds_service.py
Comment (madereddy) [reply prowler/providers/aws/services/rds/rds_service.py:173]: Switched to more closely match db_instances. Can switch it back and rework the PR if this is required.

File: prowler/providers/aws/services/rds/rds_service.py
Comment (jfagoagas) [reply prowler/providers/aws/services/rds/rds_service.py:17]: We must use maps to store objects using the ARN as key for faster retrievals.

Comment (madereddy) [reply prowler/providers/aws/services/rds/rds_service.py:17]: Ahh I understand. Im going to close this PR and redo it entirely. Thank you @jfagoagas for the guidance.

**Comment by jfagoagas** [comment]
@madereddy are you planning to re-do the PR?

**Comment by madereddy** [comment]
Yes I will redo it after the other certificate PR has been merged.

**Comment by MrCloudSec** [comment]
@madereddy I have merged the other PR!

**Comment by madereddy** [comment]
I will start working on the update commit now

**Pull request closed without merging**
